app/src/main/java/com/example/ithaca_transit_android_v2/NetworkingAdapters.kt

```diff
@@ -25,52 +31,85 @@ class LocationAdapter {
         val long: Double,
         val detail: String?
     )
+
     @FromJson
     private fun fromJson(json: DataLocation): List<Location> {
-        val finalLoc = json.data.map { loc ->
-            Location(loc.type, loc.name, Coordinate(loc.lat, loc.long), loc.detail)
+        if (json.data == null) {
+            return ArrayList()
+        }
+        val applePlaces = if(json.data.applePlaces != null) {
```

Comment:
`val applePlaces = json.data.applePlaces?.let { // code if not null } ?: { // code if null }`